include/deal.II/base/bounding_box.h

```diff
@@ -0,0 +1,104 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_base_bounding_box_h
+#define dealii_base_bounding_box_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A class that represents a bounding box in a space with arbitrary dimension
+ * <tt>spacedim</tt>.
+ *
+ * Objects of this class are used to represent bounding boxes. They are,
+ * among other uses, useful in parallel distributed meshes to give a general
+ * description the owners of each portion of the mesh.
+ *
+ * Boundig boxes are represented by two vertices (bottom left and top right).
+ * Geometrically, are represent the following:
+ * 1 d: a segment (represented by its vertices)
+ * 2 d: a rectangle (represented by two vertices V opposed to the center)
+ * .--------V
```

Comment:
Make sure you do 
~~~
@code
* .--------V
* |        |
* V--------.
@endcode
~~~

Reply:
Ok, I've slightly modified the description underlying the order must be respected (otherwise the point_inside function won't work).

Moreover there currently is an Assert which, when creating a BoundingBox from points, checks the order; do you think it'd be wiser to use an AssertThrow so that the check is done also at run time?

Reply:
Nope, `Assert` is exactly the right choice here!